### PR TITLE
Respect capabilities in automation template selection [MAILPOET-6002]

### PR DESCRIPTION
**Pull request by JanJakes**
## Description

Respect plan capabilities in automation template selection.

## Code review notes

Note that to fully test this with the shop, https://github.com/mailpoet/shop/pull/1204 must be released.

## QA notes

Note that to fully test this with the shop, https://github.com/mailpoet/shop/pull/1204 must be released.

1. Older 2002 plans should behave as before, except that purchased product/tag/category templates are now free, and there are no free-only templates (we don't hide any templates when premium is activated).
2. Newer 2024 plans should behave based on capabilities/tiers (see the specs).

Upgrade URLs should lead to `https://account.mailpoet.com/orders/upgrade/<key>?s=<number-of-subscribers>&capabilities%5BautomationSteps%5D=<number-of-required-automation-steps>`.

Please, also smoke-check other upgrade modals and banners and verify they link to correct plans.

## Linked PRs

https://github.com/mailpoet/mailpoet-premium/pull/861

## Linked tickets

_N/A_

## After-merge notes

_N/A_

## Tasks

- [ ] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [ ] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

### CODE REVIEW

**Review by brezocordero — APPROVED** [review]
@JanJakes Nice work.
The code looks good. I also tested the templates display for 2022 and 2024 keys and looked at the different modals/banners.

A couple of comments:
The condition to display the 'Thank Loyal Customers' template was a bit hard to follow. Maybe adding a comment [here](https://github.com/mailpoet/mailpoet-premium/pull/861/files#diff-464830d6573a532061dd84bfbf2eaa93267c0344b135ddc1bc3a39cd40c0b138R148) could help but it is not necessary.

I am not sure if we are planning to have [multiple capabilities in the upgrade links](https://github.com/mailpoet/mailpoet/pull/5555/commits/ebc859f28ddfb3adbd6eb615edab5080b248f66b) but if it ever happens, and more than one is restricted, the message on upgrade-info.ts will be [for the first one it finds](https://github.com/mailpoet/mailpoet/pull/5555/commits/ebc859f28ddfb3adbd6eb615edab5080b248f66b).
Unless having multiple capabilities in the upgrade links is going to happen soon, I don't think it matters.

**Comment by JanJakes** [comment]
@brezocordero 

> The condition to display the 'Thank Loyal Customers' template was a bit hard to follow. Maybe adding a comment [here](https://github.com/mailpoet/mailpoet-premium/pull/861/files#diff-464830d6573a532061dd84bfbf2eaa93267c0344b135ddc1bc3a39cd40c0b138R148) could help but it is not necessary.

Good point, thanks, added in https://github.com/mailpoet/mailpoet-premium/commit/d7f39e9342151844adc7fe5c1759d380e5a151f6.

> I am not sure if we are planning to have [multiple capabilities in the upgrade links](https://github.com/mailpoet/mailpoet/pull/5555/commits/ebc859f28ddfb3adbd6eb615edab5080b248f66b) but if it ever happens, and more than one is restricted, the message on upgrade-info.ts will be [for the first one it finds](https://github.com/mailpoet/mailpoet/pull/5555/commits/ebc859f28ddfb3adbd6eb615edab5080b248f66b).
> Unless having multiple capabilities in the upgrade links is going to happen soon, I don't think it matters.

I noticed that too, but thought it's not worth trying to universally solve now. If we ever need it, we should keep that in mind. I also noticed that the capability messages throw away any custom phrasing, which for now seems to be expected, but in the future, we may need more flexibility.

**Review by Aschepikov — APPROVED** [review]
GJ!
